src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1436,7 +1436,7 @@ protected void AfterMembersChecks(DiagnosticBag diagnostics)
 
                 // https://github.com/dotnet/roslyn/issues/30080: Report diagnostics for base type and interfaces at more specific locations.
                 var baseType = BaseTypeNoUseSiteDiagnostics;
-                var interfaces = InterfacesNoUseSiteDiagnostics();
+                var interfaces = AllInterfacesNoUseSiteDiagnostics;
```

Comment:
>AllInterfacesNoUseSiteDiagnostics [](start = 33, length = 33)

I think this wouldn't be correct for classes. Let's try to come up with a scenario when we can observe a negative effect. I think we should reuse GetInterfacesToEmit method here. This is exactly what we are looking for - the set of types we are going to emit.